spec/services/waste_exemptions_engine/edit_completion_service_spec.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+module WasteExemptionsEngine
+  RSpec.describe EditCompletionService do
+    describe "run" do
+      let(:edit_registration) { create(:edit_registration, :modified) }
+      let(:registration) do
+        Registration.where(reference: edit_registration.reference).first
+      end
+      let(:service) { EditCompletionService.run(edit_registration: edit_registration) }
+
+      skipped_attributes = %w[registration_id
+                              transient_registration_id
+                              created_at
+                              updated_at
+                              id]
+
+      (Helpers::ModelProperties::REGISTRATION - %i[reference submitted_at]).each do |attribute|
+        it "updates the registration data for #{attribute}" do
+          old_value = registration[attribute]
+          new_value = edit_registration[attribute]
+
+          expect { service }.to change {
+            registration.reload[attribute]
+          }.from(old_value).to(new_value)
+        end
+      end
+
+      %i[operator_address contact_address site_address].each do |address_type|
+        it "copies the #{address_type} from the registration" do
+          old_attributes = registration.send(address_type).attributes.except(*skipped_attributes)
+          new_attributes = edit_registration.send(address_type).attributes.except(*skipped_attributes)
+
+          expect { service }.to change {
+            registration.reload.send(address_type).attributes.except(*skipped_attributes)
+          }.from(old_attributes).to(new_attributes)
+        end
+      end
+
+      it "copies the people from the registration" do
+        old_people_data = registration.people.map do |p|
```

Comment:
[NIT] Can we rename the variable `p` to `person`? Mostly because `p` is not `free` to be used as it used as a shortcut for the `puts` method

Reply:
Oops, yes! Fixed in https://github.com/DEFRA/waste-exemptions-engine/pull/139/commits/f4a0f301cadaa71ea927d7d134c8d2f552f0bb08